File: python-selenium/web_driver_utility.py
import os

from selenium import webdriver

class WebDriverUtility():
    def get_new_web_driver(self):
        os.environ['MOZ_HEADLESS'] = '1'
        driver = webdriver.Firefox()

        return driver

    def get_home_page_url(self):
        return "http://localhost:8082"

    def get_add_car_page_url(self):
        return self.get_home_page_url() + "/addcar"

    def get_charts_page_url(self):
        return self.get_home_page_url() + "/charts"

File: java-car-data/car-data-web-ui-tests/src/test/java/com/glicerial/samples/cardata/web/uitests/WebDriverUtility.java
package com.glicerial.samples.cardata.web.uitests;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxBinary;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.firefox.FirefoxOptions;

public class WebDriverUtility {

    public WebDriver getNewWebDriver() {
        // May need following line if geckodriver not in your path
        // System.setProperty("webdriver.gecko.driver", "/usr/local/bin");

        FirefoxBinary firefoxBinary = new FirefoxBinary();
        firefoxBinary.addCommandLineOptions("--headless");
        FirefoxOptions firefoxOptions = new FirefoxOptions();
        firefoxOptions.setBinary(firefoxBinary);
        WebDriver webDriver = new FirefoxDriver(firefoxOptions);

        return webDriver;
    }

    public String getHomePage() {
        return "http://localhost:8082";
    }

    public String getAddCarPage() {
        return getHomePage() + "/addcar";
    }
}
